core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -1167,6 +1167,22 @@ class DynamicRemoteLogConfig(server: KafkaBroker) extends BrokerReconfigurable w
           throw new ConfigException(s"$errorMsg, value should be at least 1")
         }
       }
+
+      if (RemoteLogManagerConfig.REMOTE_LOG_MANAGER_COPIER_THREAD_POOL_SIZE_PROP.equals(k) ||
+        RemoteLogManagerConfig.REMOTE_LOG_MANAGER_EXPIRATION_THREAD_POOL_SIZE_PROP.equals(k) ||
+        RemoteLogManagerConfig.REMOTE_LOG_READER_THREADS_PROP.equals(k)) {
+        val newValue = v.asInstanceOf[Int]
+        val oldValue = server.config.getInt(k)
+        if (newValue != oldValue) {
+          val errorMsg = s"Dynamic thread count update validation failed for $k=$v"
+          if (newValue <= 0)
```

Comment:
The configuration can be updated to its default value of -1 through a static configuration update.

Is this an intended change that does not allow updating it to the default value (-1) through dynamic configs?

Reply:
I guess the only scenario where the user might want to set it back is if they want all three thread pools to follow the changes to `REMOTE_LOG_MANAGER_THREAD_POOL_SIZE_PROP` simultaneously.

Therefore, if we allow the user to do this, I think we should also make `REMOTE_LOG_MANAGER_THREAD_POOL_SIZE_PROP` dynamically configurable.

What are your thoughts on this?

Reply:
> The configuration can be updated to its default value of -1 through a static configuration update.

This is allowed. Only the newValue which is being dynamically updated to -1 is not allowed. 

> Is this an intended change that does not allow updating it to the default value (-1) through dynamic configs?

Yes, we made those thread-pools default to -1 to derive the value from `REMOTE_LOG_MANAGER_THREAD_POOL_SIZE_PROP`. Once those properties are configured to be a non-negative number, then we expect the user to provide subsequent valid value when updating it dynamically.

Reply:
I suggested to keep -1 for falling back to the default value in the original [PR](https://github.com/apache/kafka/pull/17499/#discussion_r1804119567). 

I am fine with the approach of not allowing back to setting default as -1 for the dynamic configuration.  Please add a java doc comment and also add a clear message in the exception for user.

Reply:
Removed the validation for copier and expiration thread pool. It follows the default validation set in ConfigDef (Value can be -1 or greater than 0). With this, the static and dynamic validations are same.